### PR TITLE
Update how-to-use-minio-server-side-encryption-with-aws-cli.md

### PR DESCRIPTION
**Pull request by jamesneimanconsulting**
@deekoder , @harshavardhana Kindly review and merge if approved

### CODE REVIEW

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 1]
```diff
@@ -1,88 +1,94 @@
...
-    ```
-    You should use your own encryption key.
-
-### 2.2 Show object information
-  You **must** specify the correct SSE-C key of an encrypted object to show its metadata:
-  ```
-  aws s3api head-object \
+# How to use Minio's Server-side Encryption with the AWS CLI
+
+Minio supports S3 server-side-encryption with customer provided keys (SSE-C). This following sections describe this use of this encryption with the AWS Command Line Interface (`awscli`):
```
Comment (eco-minio) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: "This following sections describe this use of this encryption"
The word "this: is used 3 times, describe should be describes

Comment (deekoder) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: ditto. This following sections... does not sound right.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 2]
```diff
@@ -1,88 +1,94 @@
...
+* [Prerequisites](#prerequisites)
+* [Use SSE-C with aws-cli](#use-sse-c-with-aws-cli)
+* [Security Notice](#security-notice)
+
+# <a name="prerequisites"></a>Prerequisites
+
+A client must specify three HTTP headers for SSE-C requests:
+* `X-Amz-Server-Side-Encryption-Customer-Algorithm`: The algorithm identifier: must be set to `AES256`.
+* `X-Amz-Server-Side-Encryption-Customer-Key`: The secret encryption key: must be a 256-bit Base64-encoded string.
+* `X-Amz-Server-Side-Encryption-Customer-Key-MD5`: The encryption key MD5 checksum: must be set to the MD5-sum of the encryption key. Note: The MD5 checksum is the MD5 sum of the raw binary key, not of the base64-encoded key.
```
Comment (eco-minio) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: I wasn't clear for these lines, when `:` is used after "identifier:", "key:" and "checksum:", what is this denoting?

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 3]
```diff
@@ -1,88 +1,94 @@
...
+
+```sh
+aws s3api put-object \
+  --no-verify-ssl \
+  --endpoint-url https://localhost:9000 \
+  --bucket my-bucket --key my-secret-diary \
+  --sse-customer-algorithm AES256 \
+  --sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
+  --sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+  --body ~/my-diary.txt You should use your own encryption key.
```
Comment (eco-minio) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: "...You should use your own encryption key."  This should be outside the code block

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 4]
```diff
@@ -1,88 +1,94 @@
...
+* [Security Notice](#security-notice)
+
+# <a name="prerequisites"></a>Prerequisites
+
+A client must specify three HTTP headers for SSE-C requests:
+* `X-Amz-Server-Side-Encryption-Customer-Algorithm`: The algorithm identifier: must be set to `AES256`.
+* `X-Amz-Server-Side-Encryption-Customer-Key`: The secret encryption key: must be a 256-bit Base64-encoded string.
+* `X-Amz-Server-Side-Encryption-Customer-Key-MD5`: The encryption key MD5 checksum: must be set to the MD5-sum of the encryption key. Note: The MD5 checksum is the MD5 sum of the raw binary key, not of the base64-encoded key.
+
+Install the Minio Server with TLS as described [here](https://docs.minio.io/docs/how-to-secure-access-to-minio-server-with-tls).
```
Comment (deekoder) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: I like the numbered 1/2/3 steps format.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 5]
```diff
@@ -1,88 +1,94 @@
...
+--endpoint-url https://localhost:9000 \
+--bucket my-bucket \
+--key my-secret-diary \
+--sse-customer-algorithm AES256 \
+--sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
+--sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+~/my-diary.txt
+```
+
+# <a name="security-notice"></a> Security-Related Notes
```
Comment (deekoder) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: I like that we moved this to the bottom after the steps.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 6]
```diff
@@ -1,88 +1,94 @@
...
+Delete your local copy of `my-diary.txt`:
+
+```sh
+rm ~/my-diary.txt
+```
+
+Restore the file by downloading it from the server:
+
+```sh
+Copyaws s3api get-object \
```
Comment (eco-minio) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: Is Copyaws correct here or should this just be `aws`?

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 7]
```diff
@@ -1,88 +1,94 @@
...
+--sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+~/my-diary.txt
+```
+
+# <a name="security-notice"></a> Security-Related Notes
+
+* The Minio server will reject any SSE-C request made over an insecure (non-TLS) connection as per the S3 specification the. This means that SSE-C requires TLS / HTTPS, and an SSE-C request contains the encryption key. 
+* If an SSE-C request is made over a non-TLS connection, the SSE-C encryption key must be treated as compromised.
+* Per the S3 specification, the `content-md5` returned by an SSE-C PUT operation does not match the MD5 sum of the uploaded object. 
+* The Minio server uses a tamper-proof encryption scheme to encrypt objects and does not save the encryption key, which means you are responsible for managing encryption keys. If you lose the encryption key for an object, you will lose that object.
```
Comment (eco-minio) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: "...you will lose the ability to decrypt that object"

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 8]
```diff
@@ -1,88 +1,94 @@
...
+--key my-secret-diary \
+--sse-customer-algorithm AES256 \
+--sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
+--sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+~/my-diary.txt
+```
+
+# <a name="security-notice"></a> Security-Related Notes
+
+* The Minio server will reject any SSE-C request made over an insecure (non-TLS) connection as per the S3 specification the. This means that SSE-C requires TLS / HTTPS, and an SSE-C request contains the encryption key.
```
[the remark this answers is not in the document]
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @aead Fixed. Line 89 now reads as follows:

* The Minio server will reject any SSE-C request made over an insecure (non-TLS) connection per the S3 specification. This means that SSE-C requires TLS / HTTPS, and an SSE-C request contains the encryption key. 

Apologies for the errors. I have resolved this by updating the branch. Please see https://github.com/jamesneimanconsulting/cookbook/blob/patch-1/docs/how-to-use-minio-server-side-encryption-with-aws-cli.md.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 1 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @eco-minio , @deekoder :  Fixed. Line 3 now reads as follows:

Minio supports S3 server-side-encryption with customer provided keys (SSE-C). The following sections describe the use of server-side encryption with the AWS Command Line Interface (`awscli`):

Apologies for the errors. I have resolved this by updating the branch. Please see https://github.com/jamesneimanconsulting/cookbook/blob/patch-1/docs/how-to-use-minio-server-side-encryption-with-aws-cli.md.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 2 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @eco-minio Fixed. I have divided these into shorter sentences for clarity. Lines 10-13 now read as follows:

A client must specify three HTTP headers for SSE-C requests:
* `X-Amz-Server-Side-Encryption-Customer-Algorithm`: The algorithm identifier. It must be set to `AES256`.
* `X-Amz-Server-Side-Encryption-Customer-Key`: The secret encryption key. It must be a 256-bit Base64-encoded string.
* `X-Amz-Server-Side-Encryption-Customer-Key-MD5`: The encryption key MD5 checksum. It must be set to the MD5-sum of the encryption key. Note: The MD5 checksum is the MD5 sum of the raw binary key, not of the base64-encoded key.

 I have resolved this by updating the branch. Please see https://github.com/jamesneimanconsulting/cookbook/blob/patch-1/docs/how-to-use-minio-server-side-encryption-with-aws-cli.md.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 4 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @deekoder I have fixed line 3. Please let me know if other changes are desired.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 6 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @eco-minio Revising to just `aws`.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 7 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @eco-minio Revised as suggested.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md (hunk 3 above)
Comment (jamesneimanconsulting) [reply docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: @eco-minio   I have revised as follows: 

The following example shows how to upload an object named `my-secret-diary` where the content is the file `~/my-diary.txt`. Note that you should use your own encryption key.

```sh
aws s3api put-object \
  --no-verify-ssl \
  --endpoint-url https://localhost:9000 \
  --bucket my-bucket --key my-secret-diary \
  --sse-customer-algorithm AES256 \
  --sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
  --sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
  --body ~/my-diary.txt 
```

**Comment by jamesneimanconsulting** [comment]
@deekoder I've revised the document in this same branch per the comments.

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 9]
```diff
@@ -1,88 +1,94 @@
...
+--endpoint-url https://localhost:9000 \
+--bucket my-bucket \
+--key my-secret-diary \
+--sse-customer-algorithm AES256 \
+--sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
+--sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+~/my-diary.txt
+```
+
+## <a name="security-notice"></a>3. Security-Related Notes
```
Comment (deekoder) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: James can we do this in markdown itself? Can it support name= in md?

File: docs/how-to-use-minio-server-side-encryption-with-aws-cli.md [hunk 10]
```diff
@@ -1,88 +1,94 @@
...
-  ```
-  aws s3api head-object \
+# How to use Minio's Server-side Encryption with the AWS CLI
+
+Minio supports S3 server-side-encryption with customer provided keys (SSE-C). The following sections describe the use of server-side encryption with the AWS Command Line Interface (`aws-cli`):
+* [Prerequisites](#prerequisites)
+* [Use SSE-C with aws-cli](#use-sse-c-with-aws-cli)
+* [Security-Related Notes](#security-notice)
+
+## <a name="prerequisites"></a>1. Prerequisites
```
Comment (deekoder) [inline docs/how-to-use-minio-server-side-encryption-with-aws-cli.md]: james i had a chance to think about this. Can we do this in markdown itself? Can it support name= in md?

**Comment by jamesneimanconsulting** [comment]
@deekoder I am not authorized to merge.

**Pull request merged**